packages/base-controller/src/ControllerMessenger.ts

```diff
@@ -533,7 +565,7 @@ export class ControllerMessenger<
       this.events.set(eventType, subscribers);
     }
 
-    subscribers.set(handler, selector);
+    subscribers.set(handler, selector as ExtractEventHandler<Event, E>);
```

Comment:
Highlighting for closer review.

Reply:
Hmm, will have to think about this some more.

Reply:
Updated this type here: https://github.com/MetaMask/core/pull/1890/commits/b1849805ac89323bc3f3460706281bb75f3ead46

Reply:
Continuing thread here: https://github.com/MetaMask/core/pull/1890#discussion_r1388479681